### ISSUE TITLE
`az ad sp create` not idempotent

### ISSUE THREAD

**Issue by Morriz** [Azure/azure-cli#11563]
## Describe the bug
if sp exists, the secret gets changed, resulting in downstream failures .

**Command Name**
`az ad sp create-for-rbac --skip-assignment --name $AZURE_AKS_SP_NAME --years $YEARS`

**Errors:**
```
Azure Error: InvalidTemplateDeployment
Message: The template deployment 'deploy-k8s-xxx' is not valid according to the validation procedure. The tracking id is '325a41a0-972a-41df-a47e-2781d0eadb91'. See inner errors for details.
Exception Details:
        Error Code: PropertyChangeNotAllowed
        Message: Provisioning of resource(s) for container service k8s-xxx in resource group k8s-xxx failed. Message: {
  "code": "PropertyChangeNotAllowed",
  "message": "Changing property 'servicePrincipalProfile.clientId' is not allowed.",
  "target": "servicePrincipalProfile.clientId"
 }.
```

## To Reproduce:
1 create the sp
2 create a cluster
3 create the sp again
4 create the cluster again

## Expected Behavior
no errors and to re-use the SP as is, without changing it's secret

## Environment Summary
```
Darwin-19.0.0-x86_64-i386-64bit
Python 3.7.4
Shell: bash

azure-cli 2.0.75 *

```

**Comment by haroldrandom** [comment]
@jiasli  Please take a look

**Comment by arrownj** [comment]
Hi @Morriz , this is a by design feature. And I think there are some reasons to design as this. The password is auto generated when creating the sp and return to user only once. There is no way to get this password from any API call. So if we keep it same with before when creating the sp for the second time, previous password would not be able to return to user. Also there are some other fields(startdate, enddate) in password credential would not get updated.

For you case, we suggest you to list/show the sp first. Create a new sp only when it doesn't exist.

**Comment by yungezz** [comment]
hi @Morriz does above infro from @arrownj make sense for you? would like to close the issue for now. Feel free to create new issue if met any other error. thanks.

**Comment by Morriz** [comment]
I understand. So be it
